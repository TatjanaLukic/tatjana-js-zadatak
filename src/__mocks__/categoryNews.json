{
    "status": "ok",
    "totalResults": 5,
     "articles": [
     {
     "source": {
    "id": "bild",
    "name": "Bild"
    },
    "author": "Bild.de",
    "title": "Tesla Aktie schmiert ab   Kein Saft am Batterie Tag   BILD",
    "description": "Tesla Chef Elon Musk hat vor einer Veranstaltung zur Batterie Technologie die Erwartungen gedämpft und so die Aktienhändler enttäuscht.",
    "url": "https://www.bild.de/geld/wirtschaft/wirtschaft/tesla aktie schmiert ab kein saft am batterie tag 73031786.bild.html",
    "urlToImage": "https://bilder.bild.de/fotos/tesla aktie schmiert ab kein saft am batterie tag 201524804 73032638/Bild/3.bild.jpg",
    "publishedAt": "2020 09 22T06:04:00Z",
    "content": "Stell dir vor, es ist Batterie Tag   aber es gibt nicht genug Batterien.\r\nTesla Chef Elon Musk (49) hat vor einer Veranstaltung zur Batterie Technologie die Erwartungen gedämpft und damit die Aktienh… [+1496 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "N tv.de"
    },
    "author": "n tv NACHRICHTEN",
    "title": "Strom selber machen? :Die fünf größten Irrtümer bei Solaranlagen   n tv NACHRICHTEN",
    "description": "Eigenheimbesitzer kommen am Thema Fotovoltaik kaum noch vorbei   schließlich soll eine Solaranlage nicht nur umweltfreundlich, sondern auch lukrativ sein. Wer sich über Solaranlagen informiert, stößt jedoch schnell auf widersprüchliche Aussagen. Ein Faktenche…",
    "url": "https://www.n tv.de/ratgeber/Die fuenf groessten Irrtuemer bei Solaranlagen article22040579.html",
    "urlToImage": "https://bilder3.n tv.de/img/incoming/crop14785186/9961323629 cImg_16_9 w1200/imago61527841h.jpg",
    "publishedAt": "2020 09 22T05:03:57Z",
    "content": "Eigenheimbesitzer kommen am Thema Fotovoltaik kaum noch vorbei   schließlich soll eine Solaranlage nicht nur umweltfreundlich, sondern auch lukrativ sein. Wer sich über Solaranlagen informiert, stößt… [+5128 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "Aerotelegraph.com"
    },
    "author": null,
    "title": "Warum Airbus' Wasserstoffflieger hinten keine Fenster haben   aeroTELEGRAPH",
    "description": "Zwei der drei Wasserstoffflieger, die Airbus präsentiert, unterscheiden sich auf den ersten Blick nicht sehr von heutigen Flugzeugen. Ein wichtiges Detail ist jedoch anders.",
    "url": "https://www.aerotelegraph.com/wasserstoffflieger warum airbus neue flieger hinten keine fenster haben",
    "urlToImage": "https://cdn.aerotelegraph.com/production/uploads/2020/09/airbus_zero_e.jpg",
    "publishedAt": "2020 09 22T04:55:00Z",
    "content": null
    },
     {
     "source": {
    "id": null,
    "name": "Express.de"
    },
    "author": null,
    "title": "Höhle der Löwen: Kölner Gründer klärt auf, Investoren geschockt   EXPRESS",
    "description": "„Die Höhle der Löwen“ Investoren interessieren sich sehr für das Kölner Start up „Forest Gum“. Aber reicht es am Montag für einen Deal?",
    "url": "https://www.express.de/news/promi und show/dhdl  loewen  buhlen um  forest gum   gruender schockt mit infos zu beliebtem produkt 37377432",
    "urlToImage": "https://www.express.de/image/37378240/2x1/940/470/a89bb94d2ab24631545ad42855d9b08f/NT/forest gum thomas kraemer.jpg",
    "publishedAt": "2020 09 22T03:17:00Z",
    "content": "Köln  \r\nEin kleines Unternehmen aus Köln hat Großes vor: Wir möchten den Kaugummi Markt revolutionieren, sagt Thomas Krämer. Dazu wagt sich der Gründer von Forest Gum am Montagabend (Folge vier) in D… [+5070 chars]"
    },
     {
     "source": {
    "id": "spiegel online",
    "name": "Spiegel Online"
    },
    "author": "Thomas Geiger",
    "title": "Erste Fahrt im VW ID.4: Als Volkswagen einmal schneller als Tesla war   DER SPIEGEL",
    "description": "Mit Tamtam stampft Tesla in Brandenburg die Fabrik fürs SUV Model Y aus dem Boden. Weniger beachtet: Bei VW rollen die ersten Exemplare des Konkurrenten ID.4 schon vom Band. Muss Elon Musk zittern? Eine Probefahrt.",
    "url": "https://www.spiegel.de/auto/fahrberichte/vw id 4 im test kann es der elektro suv von volkswagen mit tesla aufnehmen a f3d41599 9ab1 489e 99aa 4dc93b7f0b85",
    "urlToImage": "https://cdn.prod.www.spiegel.de/images/1c62a093 2f1a 41de a462 781e87c69868_w1280_r1.77_fpx33.34_fpy50.jpg",
    "publishedAt": "2020 09 22T02:58:38Z",
    "content": null
    } ]
    }