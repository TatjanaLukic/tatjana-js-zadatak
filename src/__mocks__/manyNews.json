{
    "status": "ok",
    "totalResults": 20,
     "articles": [
     {
     "source": {
    "id": null,
    "name": "CNET"
    },
    "author": "Jennifer Bisset",
    "title": "Emmys 2020: From Watchmen to Zendaya, the full list of winners   CNET",
    "description": "Watchmen wins best limited series, with Regina King, Zendaya and Mark Ruffalo winning big acting awards, and Schitt's Creek sweeping the comedy categories.",
    "url": "https://www.cnet.com/news/emmys 2020 from watchmen to zendaya the full list of winners/",
    "urlToImage": "https://cnet2.cbsistatic.com/img/E6TjO7eJ19Jbx0H 3RSG8AAzy10=/1200x630/2020/09/22/baee267f 0f65 4fae ad10 1d4871dff3a5/zendaya euphoria emmy nomination.jpg",
    "publishedAt": "2020 09 22T05:49:00Z",
    "content": "Zendaya in Euphoria.\r\nHBO\r\nThe 2020 Primetime Emmy Awards are done, with Schitt's Creek sweeping the comedy awards and HBO's Watchmen and Succession nabbing a host of their own big wins in the limite… [+4818 chars]"
    },
     {
     "source": {
    "id": "cnn",
    "name": "CNN"
    },
    "author": "Steven Jiang, CNN",
    "title": "Chinese tycoon who criticized Xi Jinping's handling of coronavirus jailed for 18 years   CNN",
    "description": "A Chinese billionaire who criticized President Xi Jinping's handling of the coronavirus pandemic has been jailed for 18 years on corruption charges, a court said Tuesday.",
    "url": "https://www.cnn.com/2020/09/22/asia/china ren zhiqiang xi jinping intl hnk/index.html",
    "urlToImage": "https://cdn.cnn.com/cnnnext/dam/assets/200408095601 ren zhiqiang super tease.jpg",
    "publishedAt": "2020 09 22T05:09:00Z",
    "content": "(CNN)A Chinese billionaire who criticized President Xi Jinping's handling of the coronavirus pandemic has been jailed for 18 years on corruption charges, a court said Tuesday. \r\nRen Zhiqiang, a retir… [+3410 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "Daily Mail"
    },
    "author": "By Megan Sheets For Dailymail.com",
    "title": "Lindsey Graham says Trump 'has the votes' to confirm his Supreme Court nominee   Daily Mail",
    "description": "South Carolina Sen Lindsey Graham expressed confidence in Trump's chances of rushing through a Supreme Court nominee on Monday.",
    "url": "https://www.dailymail.co.uk/news/article 8758511/Lindsey Graham says Trump votes confirm Supreme Court nominee.html",
    "urlToImage": "https://i.dailymail.co.uk/1s/2020/09/22/05/33466268 0 image a 29_1600749926139.jpg",
    "publishedAt": "2020 09 22T04:38:09Z",
    "content": "Senate Judiciary Committee Chairman Lindsey Graham declared that Donald Trump 'has the votes' to confirm his Supreme Court nominee after two key Republican swing voters voiced their support for the p… [+35047 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "Raiders.com"
    },
    "author": null,
    "title": "Derek Carr leaves his mark on Allegiant Stadium forever with a signature win   Raiders.com",
    "description": "The Las Vegas Raiders quarterback made sure to let everyone know exactly whose stadium this is after sending the Saints home with a loss.",
    "url": "https://www.raiders.com/news/derek carr leaves his mark on allegiant stadium forever with a signature win",
    "urlToImage": "https://static.clubs.nfl.com/image/private/t_editorial_landscape_12_desktop/raiders/an9m7hwqiwghh5jxjxzi",
    "publishedAt": "2020 09 22T04:27:38Z",
    "content": "It has become a tradition to leave the result of a stadium's first game etched on it forever. Eli Manning infamously signed the concrete outside of the visitor's locker in AT&amp;T Stadium after beat… [+384 chars]"
    },
     {
     "source": {
    "id": "the verge",
    "name": "The Verge"
    },
    "author": "Sean Hollister",
    "title": "Watch a demo of Royole's new folding phone, on sale today for roughly $1500   The Verge",
    "description": "The world’s first commercially available folding screen phone wasn’t made by Samsung, Huawei or Motorola — it was the Royole Flexpai, a device that we called “charmingly awful.” But in March, Royole promised to refine the idea with the new FlexPai 2, and the …",
    "url": "https://www.theverge.com/2020/9/22/21450343/royole flexpai 2 folding phone price specs release date china",
    "urlToImage": "https://cdn.vox cdn.com/thumbor/SgwLJ50yqvGDOQiPCzqUXYxYPRI=/0x70:1920x1075/fit in/1200x630/cdn.vox cdn.com/uploads/chorus_asset/file/21900889/flexpai_2_render.jpg",
    "publishedAt": "2020 09 22T04:24:52Z",
    "content": "The FlexPai 2 is launching in China\r\nImage: Royole\r\nThe worlds first commercially available folding screen phone wasnt made by Samsung, Huawei or Motorola it was the Royole Flexpai, a device that we … [+2443 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "LAist"
    },
    "author": null,
    "title": "Bobcat Fire: Fire Burns Through More Than 106K Acres; Flames Again Threaten Mt. Wilson   LAist",
    "description": "The fire is now one of the largest wildfires in Los Angeles County history.",
    "url": "https://laist.com/latest/post/20200921/bobcat fire day16",
    "urlToImage": "https://laistassets.scprdev.org/i/9320163e4322186bfed7ee52367109ca/5f697b1dedff0a0008ef3d30 eight.jpg",
    "publishedAt": "2020 09 22T04:16:00Z",
    "content": "This story is no longer being actively updated. For the latest overnight updates, check the following official sites and social channels:\r\nAngeles National Forest Facebook | Angeles National Forest T… [+8660 chars]"
    },
     {
     "source": {
    "id": "google news",
    "name": "Google News"
    },
    "author": null,
    "title": "Xbox/Bethesda Deal: What It Means For PS5   IGN Daily Fix   IGN",
    "description": null,
    "url": "https://news.google.com/__i/rss/rd/articles/CBMiK2h0dHBzOi8vd3d3LnlvdXR1YmUuY29tL3dhdGNoP3Y9WjJxeWZhU2J1SU3SAQA?oc=5",
    "urlToImage": null,
    "publishedAt": "2020 09 22T04:15:00Z",
    "content": null
    },
     {
     "source": {
    "id": null,
    "name": "Phys.Org"
    },
    "author": "Science X staff",
    "title": "Evolution of radio resistance is more complicated than previously thought   Phys.org",
    "description": "The toughest organisms on Earth, called extremophiles, can survive extreme conditions like extreme dryness (desiccation), extreme cold, space vacuum, acid, or even high level radiation. So far, the toughest of all seems to be the bacterium Deinococcus radiodu…",
    "url": "https://phys.org/news/2020 09 evolution radio resistance complicated previously thought.html",
    "urlToImage": "https://scx2.b cdn.net/gfx/news/hires/2020/evolutionofr.jpg",
    "publishedAt": "2020 09 22T04:10:02Z",
    "content": "The toughest organisms on Earth, called extremophiles, can survive extreme conditions like extreme dryness (desiccation), extreme cold, space vacuum, acid, or even high level radiation. So far, the t… [+3528 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "BBC News"
    },
    "author": "https://www.facebook.com/bbcnews",
    "title": "Asian stock markets continue the global fallout   BBC News",
    "description": "Australian shares were among the hardest hit in the region falling to a three month low.",
    "url": "https://www.bbc.com/news/business 54230742",
    "urlToImage": "https://ichef.bbci.co.uk/news/1024/branded_news/181A/production/_114507160_asx.jpg",
    "publishedAt": "2020 09 22T03:26:26Z",
    "content": "Image copyrightGetty Images\r\nStock markets in Asia suffered on Tuesday from the fallout from UK and US investors worried about a rise in coronavirus cases.\r\nThe biggest falls in the region were recor… [+1665 chars]"
    },
     {
     "source": {
    "id": "google news",
    "name": "Google News"
    },
    "author": null,
    "title": "iOS 14   10 Widget Apps You Should Try!   9to5Mac",
    "description": null,
    "url": "https://news.google.com/__i/rss/rd/articles/CBMiK2h0dHBzOi8vd3d3LnlvdXR1YmUuY29tL3dhdGNoP3Y9ampmWTVKbXkzNlHSAQA?oc=5",
    "urlToImage": null,
    "publishedAt": "2020 09 22T03:11:05Z",
    "content": null
    },
     {
     "source": {
    "id": "cnn",
    "name": "CNN"
    },
    "author": "Amir Vera and Jill Martin, CNN",
    "title": "Michael Jordan and Denny Hamlin team up to start NASCAR team, with Bubba Wallace as a driver   CNN",
    "description": "Jordan becomes the first Black majority owner of a full time race team in NASCAR's premier series in 60 years.",
    "url": "https://www.cnn.com/2020/09/21/us/michael jordan denny hamlin bubba wallace nascar spt trnd/index.html",
    "urlToImage": "https://cdn.cnn.com/cnnnext/dam/assets/200921213452 jordan hamlin wallace split restricted super tease.jpg",
    "publishedAt": "2020 09 22T03:10:00Z",
    "content": null
    },
     {
     "source": {
    "id": "nbc news",
    "name": "NBC News"
    },
    "author": "Tim Stelloh",
    "title": "Bodycam footage shows Utah police shooting teen after mom called 911 to have him hospitalized   NBC News",
    "description": "Authorities in Salt Lake City released body camera footage Monday that shows an officer shooting a 13 year old boy after his mother called 911 because the teen was acting out and needed help.",
    "url": "https://www.nbcnews.com/news/us news/bodycam footage shows utah police shooting teen after mom called n1240673",
    "urlToImage": "https://media2.s nbcnews.com/j/newscms/2020_39/3413856/200921 salt lake city police body cam snip ac 1001p_8a4ed05cd101fcae9fa98479d43ef7b8.nbcnews fp 1200 630.jpg",
    "publishedAt": "2020 09 22T02:52:00Z",
    "content": "Authorities in Salt Lake City released body camera footage Monday that shows an officer shooting a 13 year old boy after his mother called 911 because the teen was acting out and needed help.\r\nIn the… [+3177 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "Rolling Stone"
    },
    "author": "Ryan Reed",
    "title": "Avril Lavigne Lines Up Lyme Disease Livestream Benefit Show   Rolling Stone",
    "description": "Singer's site describes virtual event as \"only chance\" to see her perform in 2020",
    "url": "https://www.rollingstone.com/music/music news/avril lavigne lyme disease livestream benefit show 1064561/",
    "urlToImage": "https://www.rollingstone.com/wp content/uploads/2020/09/AP_19306514883282.jpg",
    "publishedAt": "2020 09 22T02:24:00Z",
    "content": "Avril Lavigne announced a livestream benefit concert for October 24th at 8 p.m. ET, helping raise awareness and funds in the fight against Lyme disease. Full proceeds from the virtual event — describ… [+1617 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "TMZ"
    },
    "author": "TMZ Staff",
    "title": "Sylvester Stallone's Mother, Jackie Stallone, Dead at 98   TMZ",
    "description": "Jackie Stallone has passed away at the age of 98.",
    "url": "https://www.tmz.com/2020/09/21/sylvester stallone mother mom jackie dead dies/",
    "urlToImage": "https://imagez.tmz.com/image/c5/16by9/2020/09/22/c56b017785c748298af867ad7cbc286c_xl.jpg",
    "publishedAt": "2020 09 22T02:11:00Z",
    "content": "Jackie Stallone, the famed celeb astrologer and Sly's mother, has died ... TMZ has learned.\r\nJackie recently passed away ... according to sources with direct knowledge. The exact circumstances surrou… [+1462 chars]"
    },
     {
     "source": {
    "id": "cnn",
    "name": "CNN"
    },
    "author": "Ariane de Vogue and Kevin Liptak, CNN",
    "title": "Judge on Trump's SCOTUS list volunteered for the 1987 Biden presidential campaign   CNN",
    "description": "Joan Larsen, one of the candidates on President Donald Trump's list of potential nominees for the Supreme Court, once volunteered for Joe Biden's 1987 presidential campaign.",
    "url": "https://www.cnn.com/2020/09/21/politics/joan larsen joe biden/index.html",
    "urlToImage": "https://cdn.cnn.com/cnnnext/dam/assets/200921205844 judge joan larsen super tease.jpg",
    "publishedAt": "2020 09 22T01:38:00Z",
    "content": null
    },
     {
     "source": {
    "id": "the verge",
    "name": "The Verge"
    },
    "author": "Verge Staff",
    "title": "Are Microsoft and Sony offering enough for you to buy a next gen console this year?   The Verge",
    "description": "After months of waiting, we finally know the launch details for the Xbox Series X and S and the PlayStation 5. But are Microsoft and Sony offering enough to convince you to buy into the next generation at launch? The Verge staff weighs in.",
    "url": "https://www.theverge.com/2020/9/21/21449985/xbox series x s playstation 5 ps5 next generation console upgrade preorder",
    "urlToImage": "https://cdn.vox cdn.com/thumbor/4mjTv4b y2GFub0Q4oW5Ko8uAXM=/0x162:2040x1230/fit in/1200x630/cdn.vox cdn.com/uploads/chorus_asset/file/21865121/twarren_200908_4177_0008.0.jpg",
    "publishedAt": "2020 09 22T01:21:57Z",
    "content": "After months of waiting, we finally know when the Xbox Series X, Series S and PlayStation 5 will launch, and some games that will be available on day one. \r\nBut the new hardware requires some investm… [+9646 chars]"
    },
     {
     "source": {
    "id": null,
    "name": "Vulture"
    },
    "author": "Halle Kiefer",
    "title": "The One Where Gabrielle Union Hosts an All Black Table Reading of Friends   Vulture",
    "description": "With Sterling K. Brown and Ryan Bathe taking on the roles of Ross and Rachel.",
    "url": "http://www.vulture.com/2020/09/gabrielle union to host all black table read of friends.html",
    "urlToImage": "https://pyxis.nymag.com/v1/imgs/998/e59/e4df6c7edc4e2fd71e059b961fd86945bc 21 Gabrielle Union.1x.rsocial.w1200.jpg",
    "publishedAt": "2020 09 22T01:02:00Z",
    "content": "It really hasnt been your day, your week, your month, or even your year, and honestly that isnt going to change anytime soon, but maybe an evening of Friends hosted by Gabrielle Union might help take… [+993 chars]"
    },
     {
     "source": {
    "id": "politico",
    "name": "Politico"
    },
    "author": "Nick Niedzwiadek",
    "title": "Trump: ‘I’m so angry at Republicans’   POLITICO",
    "description": "The president says his fellow Republicans have let him it down when it comes to investigating Obama and the Bidens.",
    "url": "https://www.politico.com/news/2020/09/21/trump angry republicans 419790",
    "urlToImage": "https://static.politico.com/ca/00/b33a8de44c439fcd7d83230aad5f/20200921 donaldtrump ap 773.jpg",
    "publishedAt": "2020 09 22T00:35:00Z",
    "content": "Those calls had largely gone unheeded by Republicans who control the Senate until they nabbed a foothold recently after the Senate Homeland Security and Governmental Affairs committee voted to author… [+2572 chars]"
    },
     {
     "source": {
    "id": "google news",
    "name": "Google News"
    },
    "author": null,
    "title": "CDC walks back guidance on aerosol spread of coronavirus   CBS News",
    "description": null,
    "url": "https://news.google.com/__i/rss/rd/articles/CBMiK2h0dHBzOi8vd3d3LnlvdXR1YmUuY29tL3dhdGNoP3Y9Q3JVX3RGNzdXTWvSAQA?oc=5",
    "urlToImage": null,
    "publishedAt": "2020 09 22T00:18:36Z",
    "content": null
    },
     {
     "source": {
    "id": "the washington post",
    "name": "The Washington Post"
    },
    "author": "Tony Romm",
    "title": "Justice Department expected to brief state attorneys general this week on imminent Google antitrust lawsuit   The Washington Post",
    "description": "The Justice Department is expected to brief state attorneys general this week about its imminent plans to file an antitrust lawsuit against Google, setting in motion a landmark legal clash between the U.S. government and the search and advertising behemoth.",
    "url": "https://www.washingtonpost.com/technology/2020/09/21/google antitrust state ags/",
    "urlToImage": "https://www.washingtonpost.com/wp apps/imrs.php?src=https://arc anglerfish washpost prod washpost.s3.amazonaws.com/public/MZWRIIXXHII6VBPXLFARRCUYZU.jpg&w=1440",
    "publishedAt": "2020 09 21T23:40:00Z",
    "content": "The Justice Department and Google each declined to comment.\r\nThe Justice Department could file a lawsuit against Google this month, overriding skepticism from its own top lawyers\r\nThe Justice Departm… [+3055 chars]"
    }
    ]
    }