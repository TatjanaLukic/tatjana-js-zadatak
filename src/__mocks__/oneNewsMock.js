export const oneNewsMock = {
  source: {
    id: null,
    name: "CNET",
  },
  author: "Eli Blumenthal",
  title:
    "PlayStation 5 launch confirmed: Sony's new consoles go on sale Nov. 12 starting at $400 - CNET",
  description: "We finally know how much PS5 will cost.",
  url:
    "https://www.cnet.com/news/playstation-5-launch-confirmed-sony-new-consoles-go-on-sale-nov-12-starting-at-400/",
  urlToImage:
    "https://cnet1.cbsistatic.com/img/dUl2URQq7dAyQ0mGECp2k9M5MvQ=/1200x630/2020/09/17/25d17eb4-fa01-4a22-8923-1fa1e59e1eb4/ps5-screenshot-09-16-20.png",
  publishedAt: "2020-09-17T17:02:00Z",
  content:
    'The PlayStation 5 now has a price and release date. \r\nSony\r\nReady for the PlayStation 5? You can now preorder the new console "at select retailers" for $500 (£450, AU$750) for the Blu-ray equipped ve… [+3327 chars]',
};
