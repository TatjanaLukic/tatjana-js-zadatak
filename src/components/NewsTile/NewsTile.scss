@import "../../assets/variables";

.news-tile--wrapper {
  box-sizing: border-box;
  min-width: 300px;
  height: 320px;
  width: 100%;
  background-color: white;
  overflow: hidden;
  margin-bottom: 2rem;
  border-radius: 3%;
  border: 3px solid grey;
  padding-left: 0.5rem;

  @media screen and (min-width: $smallSize) {
    width: 32%;
    height: 320px;
  }
}

.news-tile {
  padding: 0.3rem 0.5rem 0.1rem 0.3rem;
}

.news-tile__title {
  height: 50px;
  font-size: 0.9rem;
  line-height: 1.5;
  font-weight: 600;
  padding-top: 0.2rem;
  margin-top: 0.2rem;

  @media screen and (min-width: $smallSize) {
    font-size: 1rem;
    letter-spacing: 0.2px;
  }
}

.news-tile__img--wrapp {
  max-width: 330px;
  height: 170px;
  margin-right: auto;
  overflow: hidden;

  @media screen and (min-width: $smallSize) {
    max-width: 320px;
    height: 180px;
  }
}

.news-tile__img {
  max-width: 330px;
  height: auto;

  @media screen and (min-width: $smallSize) {
    max-width: 330px;
    height: 180px;
  }
}

.news-tile__desc--wrapp {
  padding: 0.1rem;
  overflow: auto;
}
.news-tile__desc {
  font-size: 78%;
  padding-top: 0.3rem;
  height: 45px;
  overflow: auto;
}

.news-tile__link {
  color: $nav-second-color;
  margin-right: 20px;
  background-color: white;
  border: none;
  font-weight: bold;
  float: right;
  text-decoration: none;
}
