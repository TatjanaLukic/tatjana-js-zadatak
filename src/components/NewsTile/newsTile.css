@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,500;0,700;1,400&display=swap");
.news-tile--wrapper {
  box-sizing: border-box;
  font-family: "Roboto", sans-serif;
  height: 320px;
  width: 100%;
  min-width: 300px;
  max-width: 400px;
  background-color: white;
  overflow: hidden;
  margin-bottom: 2rem;
  border-radius: 3%;
  border: 3px solid grey;
  padding-left: 0.5rem;
}
@media screen and (min-width: 576px) {
  .news-tile--wrapper {
    width: 32%;
    min-width: 380px;
    height: 320px;
  }
}

.news-tile {
  padding: 0.3rem 0.5rem 0.1rem 0.3rem;
}

.news-tile__title {
  height: 50px;
  font-size: 0.9rem;
  line-height: 1.6;
  font-weight: 600;
  padding: 0;
  padding-top: 0.2rem;
  padding-right: 0.5rem;
  margin: 0;
  margin-top: 0.2rem;
  position: relative;
  z-index: 1;
}
@media screen and (min-width: 576px) {
  .news-tile__title {
    font-size: 1rem;
    letter-spacing: 0.2px;
  }
}

.news-tile__img--wrapp {
  max-width: 330px;
  height: 170px;
  margin-right: auto;
  overflow: hidden;
}
@media screen and (min-width: 576px) {
  .news-tile__img--wrapp {
    max-width: 320px;
    height: 180px;
  }
}

.news-tile__img {
  max-width: 330px;
  height: auto;
}
@media screen and (min-width: 576px) {
  .news-tile__img {
    max-width: 330px;
    height: 180px;
  }
}

.news-tile__desc--wrapp {
  padding: 0.1rem;
  padding-right: 1rem;
  overflow: auto;
}

.news-tile__desc {
  padding-right: 0.5rem;
  font-size: 82%;
  line-height: 1rem;
  padding-top: 0.3rem;
  height: 45px;
}

.news-tile__link {
  color: #8e0a8e;
  background-color: white;
  border: none;
  font-weight: bold;
  float: right;
  text-decoration: none;
}

/*# sourceMappingURL=newsTile.css.map */
