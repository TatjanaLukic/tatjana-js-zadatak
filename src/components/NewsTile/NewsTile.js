import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import "./newsTile.css";

const NewsTile = ({
  title,
  urlToImage,
  description,
  author,
  content,
  publishedAt,
  parentPath,
  categoryName,
}) => {
  return (
    <div className="news-tile--wrapper" data-testid="news-tile">
      <div className="news-tile">
        <h5 className="news-tile__title">{title}</h5>
        <div className="news-tile__img--wrapp">
          <img
            className="news-tile__img"
            src={urlToImage}
            alt="pictureOfNews"
          ></img>
        </div>
        <div className="news-tile__desc--wrapp">
          <div className="news-tile__desc" data-testid="description">{description}</div>
          <Link
            className="news-tile__link"
            to={{
              pathname: `${parentPath}/NewsPage`,
              state: {
                title,
                urlToImage,
                author,
                content,
                publishedAt,
                parentPath: `${parentPath}/TopNewsPage`,
                categoryName,
              },
            }}
          >
            More &gt;
          </Link>
        </div>
      </div>
    </div>
  );
};

NewsTile.propTypes = {
  title: PropTypes.string,
  urlToImage: PropTypes.string,
  description: PropTypes.string,
  author: PropTypes.string,
  content: PropTypes.string,
  publishedAt: PropTypes.string,
};

export default NewsTile;
