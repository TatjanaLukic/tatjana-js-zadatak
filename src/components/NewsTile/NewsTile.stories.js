import React from "react";
import { MemoryRouter } from "react-router-dom";
import NewsTile from "./NewsTile";
import {oneNewsMock as mock} from "../../__mocks__/oneNewsMock";

export default {
  title: "App/NewsTile",
  component: NewsTile,
  decorators: [(getStory) => <MemoryRouter>{getStory()}</MemoryRouter>],
};

const Template = (args) => <NewsTile {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: mock.title,
  urlToImage: mock.urlToImage,
  description: mock.description,
  author: mock.author,
  content: mock.content,
  publishedAt: mock.publishedAt,
};
