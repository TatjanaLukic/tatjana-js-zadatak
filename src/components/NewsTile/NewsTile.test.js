import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NewsTile from "./NewsTile";
import { oneNewsMock as mock } from "../../__mocks__/oneNewsMock";

describe("<NewsTile />", () => {
  let newsTile;
  let data = {
    title: mock.title,
    urlToImage: mock.urlToImage,
    description: mock.description,
    author: mock.author,
    content: mock.content,
    publishedAt: mock.publishedAt,
    parentPath:"TopNewsPage"
  };

  it("should have heading,img,decription and link", () => {
    render(
      <MemoryRouter>
        <NewsTile {...data} />
      </MemoryRouter>
    );
    expect(
      screen.getByRole("heading", { name: mock.title })
    ).toBeInTheDocument();
    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(screen.getByTestId("description")).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "More >" })).toBeInTheDocument();
  });

  it("spanshot matches", () => {
    const { container } = render(
      <MemoryRouter>
        <NewsTile {...data} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
