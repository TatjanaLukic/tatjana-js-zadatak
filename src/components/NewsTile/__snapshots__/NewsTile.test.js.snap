// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsTile /> spanshot matches 1`] = `
<div>
  <div
    class="news-tile--wrapper"
    data-testid="news-tile"
  >
    <div
      class="news-tile"
    >
      <h5
        class="news-tile__title"
      >
        PlayStation 5 launch confirmed: Sony's new consoles go on sale Nov. 12 starting at $400 - CNET
      </h5>
      <div
        class="news-tile__img--wrapp"
      >
        <img
          alt="pictureOfNews"
          class="news-tile__img"
          src="https://cnet1.cbsistatic.com/img/dUl2URQq7dAyQ0mGECp2k9M5MvQ=/1200x630/2020/09/17/25d17eb4-fa01-4a22-8923-1fa1e59e1eb4/ps5-screenshot-09-16-20.png"
        />
      </div>
      <div
        class="news-tile__desc--wrapp"
      >
        <div
          class="news-tile__desc"
          data-testid="description"
        >
          We finally know how much PS5 will cost.
        </div>
        <a
          class="news-tile__link"
          href="TopNewsPage/NewsPage"
        >
          More &gt;
        </a>
      </div>
    </div>
  </div>
</div>
`;
